### ISSUE TITLE
Filenames with dots

### ISSUE THREAD

**Issue by tallesl** [python/mypy#8453]
Calling `mypy .` finds and checks successfully the files `foobar.py`, `foo-bar.py` and `foo_bar.py`, but it doesn't look up `foo.bar.py`.

Is this a bug?

At first I thought that it could be interpreting `.bar.py` as its extension, but `os.path.splitext('foo.bar.py')` gives `.py`. Tested with bash, python 3.7.6 and mypy 0761.

**Comment by ethanhs** [comment]
Huh, I seem unable to reproduce this. What OS are you on?

**Comment by hauntsaninja** [comment]
Repros on MacOS for me:
```
~/delete λ mkdir test      
~/delete λ echo 'x = 1; x += "asdf"' > test/test.lol.py
~/delete λ mypy test       
There are no .py[i] files in directory 'test'
```

**Comment by ethanhs** [comment]
Hm, thinking about this more, I'm not actually sure that this should be supported? Names with `.` in their name aren't valid modules even to CPython (if you try running them with -m). And you can't import them without using `importlib`.

@tallesl, what is your usecase?

**Comment by pr4k** [comment]
As far as i remember , its also mentioned in the official pep guidelines that using periods should be avoided in the names of module,but the question is if someone is just developing a standalone script or maybe something which is not developed as a module to be imported, should we allow it for that?
@ethanhs @tallesl @hauntsaninja

**Comment by ethanhs** [comment]
If they are developing a standalone script they should run `mypy script.dot.py`, which works in my testing.

**Comment by pr4k** [comment]
Yeah valid point! So should we then raise a error if someone tries to use period in names for a project, because right now what i have observed is that mypy ignores the file with `dot` in names when we use `mypy .`, So a response stating the ignored file could be useful i think

**Comment by tallesl** [comment]
I have a repo with configuration files by environment, like "
configuration.production.py". My deployment script then picks the desired
one and yields "configuration.py". I know python has INI support, but
they're much more painful than using pure python (but that's off-topic
anyway).

Considering that is not a valid module name, that's fair that it doesn't
find the file.

**Comment by msullivan** [comment]
An error might be a little too aggressive in some situations, though. I think maybe we say this isn't a bug?

**Comment by ethanhs** [comment]
Yeah I think it's fine to close as won'tfix.

**Comment by smheidrich** [comment]
It's great that these are excluded by default, but IMO that should be mentioned in the docs, e.g. under [Mapping file paths to modules](https://mypy.readthedocs.io/en/stable/running_mypy.html#mapping-file-paths-to-modules), because that goes into people's decisions on whether they want to just do `mypy .` in CI (and recommend doing this to their users) and what they have to put in Mypy's excludes if they do (e.g. virtualenv dirs placed inside the project dir and the like).

**Comment by hauntsaninja** [comment]
It is mentioned in the docs you link to:

> Mypy will recursively discover and check all files ending in .py or .pyi in directory paths provided, after accounting for --exclude

**Comment by smheidrich** [comment]
@hauntsaninja Sorry, I misattributed what I was seeing (Mypy ignoring e.g. `.venv` folders) to what is being described by this issue, when in fact it seems like a different matter entirely (and I got it wrong too: folders with dots in them are actually _not_ ignored _unless_ they start with a dot, in which case they are). You're right that the behavior that this issue is concerned with is already covered by the docs. I'll open a separate issue.
